### DISCUSSION TITLE
FOQUS call Simsinter created Json file: input data type 'str', need it to be 'float' type

### DISCUSSION THREAD

**Discussion by Lingyan90** [CCSI-Toolset/FOQUS#1185] [General]
I have some parameters from Aspen Plus v12 to be studied in FOQUS. The parameters from Aspen Plus `\Data\Properties\Parameters\...`
![image](https://github.com/CCSI-Toolset/FOQUS/assets/88555906/abc3668f-18cf-4d34-9725-1e45d0a79d33)
are default as `string` type when loaded in SinterCongfigGui v2.1.0.1.
![image](https://github.com/CCSI-Toolset/FOQUS/assets/88555906/bf96ba36-6135-4d37-839d-8828c0b18138)
When load the SinterCOngfigGui generated json file via `Add/Update Turbine Model` in FOQUS (TurbineLite V2.3.0.1.), the parameters stays to be `string` type. And the min, max value also a `string` type. (TurbineLite V2.3.0.1.) I didn't find anywhere allow me to change data type in Aspen Plus or SinterConfigGUI. One thing I can do is to mannually change the `string` to `float` in FOQUS, and the `min`, `max ` as well. This is doable, except that it a bit time consuming everytime need to reload the Turbine file (e.g. added more variables, parameters from Aspen), I need to change it again.

Keith@ksbeattie suggest that other way is to change the type in json file directly. Which is more efficient than change it from FOQUS everytime reload it. When open the json file in Notepad, it shows `string` type:
![image](https://github.com/CCSI-Toolset/FOQUS/assets/88555906/b269c0a6-4973-4524-858d-29cdd06d169b) and I would also need to add `min` and `max `manually here as well. Similar as a `double `type input parameter: 
![image](https://github.com/CCSI-Toolset/FOQUS/assets/88555906/9a30107d-47fb-4023-a492-de5471280e0d)
 
@kbuma suggest that it might be something that Aspen Plus support team could help. Given that in Simsinter, call parameter from Aspen does not have value shown in the Variable Explorer as shown in the first figure, while call inputs from Aspen has value shown in Variable Explorer shows below:
![image](https://github.com/CCSI-Toolset/FOQUS/assets/88555906/f95def8f-476f-45eb-a309-dde59488ad89)
I'm looking forward to further discussion of this and hope for a more convinient solution.

**Comment by jmorgan29** [comment]
@Lingyan90, note that the property parameter values are stored as string type under VAL and double type under UVAL. You can access the parameters directly by using UVAL and not have to manually change the type.

**Reply by Lingyan90** [reply]
Got ya. Just double checked and it is true that UVAL gives double type. Thanks a lot @jmorgan29 .
